pkg/sidecar/apphelper.go

```diff
@@ -14,86 +14,81 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package apphelper
+package sidecar
 
 import (
+	"bufio"
 	"fmt"
+	"io"
+	"os"
 	"time"
-
-	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
-
-	"github.com/presslabs/mysql-operator/pkg/sidecar/util"
 )
 
-var log = logf.Log.WithName("sidecar.apphelper")
-
 const (
 	// timeOut represents the number of tries to check mysql to be ready.
 	timeOut = 60
 	// connRetry represents the number of tries to connect to master server
 	connRetry = 10
 )
 
-// RunRunCommand is the main command, and represents the runtime helper that
+// RunSidecarCommand is the main command, and represents the runtime helper that
 // configures the mysql server
-func RunRunCommand(stopCh <-chan struct{}) error {
+func RunSidecarCommand(cfg *Config) error {
 	log.Info("start initialization")
 
 	// wait for mysql to be ready
-	if err := waitForMysqlReady(); err != nil {
+	if err := waitForMysqlReady(cfg); err != nil {
 		return fmt.Errorf("mysql is not ready, err: %s", err)
 	}
 
 	// deactivate super read only
 	log.Info("temporary disable SUPER_READ_ONLY")
-	if err := util.RunQuery("SET GLOBAL READ_ONLY = 1; SET GLOBAL SUPER_READ_ONLY = 0;"); err != nil {
+	if err := runQuery(cfg, "SET GLOBAL READ_ONLY = 1; SET GLOBAL SUPER_READ_ONLY = 0;"); err != nil {
```

Comment:
Can we have the defer pattern here:
```go
enableSuperReadOnly, err := disableSuperReadOnly()
defer enableSuperReadOnly()
```

Reply:
This PR does not introduce `disableSuperReadOnly` function. Only in #243 this function is added.